src/opera/compare/instance_comparer.py

```diff
@@ -24,6 +24,21 @@ def compare_topology_template(self, topology_template1,
 
         return equal, diff_copy
 
+    def prepare_update(self, topology_template1, topology_template2,
+                       template_diff):
+        # no need to take relationships into account
+        # as they are always in "initial" state
+        nodes_diff = template_diff.changed.get("nodes", Diff())
+        for node1 in topology_template1.nodes.values():
+            node_name = self._get_template_name(node1)
+            if not nodes_diff.find_key(node_name):
+                node1.set_state("initial", write=False)
```

Comment:
With these states one can easily make a mistake. I believe that we should create an enum-like python class for storing these states and call them throughout the code. We should create an issue for that. I'm just writing it down here not to forget.